lua/auto-save/init.lua

```diff
@@ -98,23 +114,49 @@ function M.save(buf)
   end
 end
 
-local save_func = nil
+function M.immediate_save()
+    local buf = api.nvim_get_current_buf()
+    cancel_timer(buf)
+    save(buf)
+end
 
-local function perform_save()
+
+local save_func = nil
+local function defer_save()
+  -- why is this needed? auto_save_abort is never set to true anyways?
+  -- TODO: remove?
   g.auto_save_abort = false
+
+  -- is it really needed to cache this function
+  -- TODO: remove?
   if save_func == nil then
-    save_func = (cnf.opts.debounce_delay > 0 and debounce(M.save, cnf.opts.debounce_delay) or M.save)
+    save_func = (cnf.opts.debounce_delay > 0 and debounce(save, cnf.opts.debounce_delay) or save)
   end
   save_func()
 end
 
 function M.on()
-  api.nvim_create_autocmd(cnf.opts.trigger_events, {
+  api.nvim_create_autocmd(cnf.opts.trigger_events.immediate_save, {
+    callback = M.immediate_save,
+    pattern = "*",
```

Comment:
Isn't `pattern = "*",` the default?

Reply:
Where do I get information about the defaults for the `nvim` api? I didn't see them in here: https://neovim.io/doc/user/api.html :(

But I think you are right with this!

Reply:
removed all `pattern = "*"` lines